src/str/transform.php

```diff
@@ -242,9 +244,96 @@ function replace_every_ci(
   );
 }
 
-function reverse(
-  string $string,
+/**
+ * Returns the "haystack" string with all occurrences of the keys of
+ * `$replacements` replaced by the corresponding values.
+ *
+ * Once a substring has been replaced, its new value will not be searched
+ * again.
+ *
+ * - For having new values searched again, see `Str\replace_every()`.
+ */
+<<__Rx>>
+function replace_every_nonrecursive(
+  string $haystack,
+  KeyedContainer<string, string> $replacements,
+): string {
+  invariant(
+    !C\contains_key($replacements, ''),
+    'Expected non-empty keys only.',
+  );
+
+  /* HH_FIXME[2049] calling stdlib directly */
+  /* HH_FIXME[4107] calling stdlib directly */
+  /* HH_FIXME[4200] Rx calling non-rx */
+  return \strtr($haystack, $replacements);
+}
+
+/**
+ * Returns the "haystack" string with all occurrences of the keys of
+ * `$replacements` replaced by the corresponding values (case-insensitive).
+ *
+ * Once a substring has been replaced, its new value will not be searched
+ * again.
+ * If a replacer is the prefix of another (like "car" and "carpet"), the longer
+ * one (carpet) gets replaced, if there is any ambiguity.
+ * When two replacers are passed that are identical except for case
+ * an invariant exception is thrown.
+ *
+ * - For having new values searched again, see `Str\replace_every_ci()`.
+ *
+ * Time complexity O(length * a) where `length` is the length of the subject
+ * and a is the amount of replacers with different lengths.
```

Comment:
This calculation seems to ignore the complexity of `slice` and `contains_key`, which could be significant for long strings. I think the correct complexity is

> O(a + length * b) where a is the sum of all key lengths and b is the sum of distinct key lengths

but you might want to check my math below, I might have missed somthing.

I would also add something like:

> This is not optimal; avoid using this function with long inputs, or where performance is critical.